content/blog/2020-01-ggplot2-3-3-0/index.Rmarkdown

```diff
@@ -0,0 +1,231 @@
+
+```{r}
+p <- ggplot(mpg) + 
+  geom_point(aes(displ, cty, size = hwy, colour = hwy))
+
+p + 
+  scale_size_binned()
+```
+
+```{r}
+p + 
+  scale_colour_binned()
+```
+
+As can be seen, the legends will reflect the binned nature of the scale by placing tick marks at the border between bins. By default the out-most ticks are not shown, indicating that the binning is open in both ends. This behavior can be controlled in the guide.
+
+```{r}
+p + 
+  scale_size_binned(guide = guide_bins(show.limits = TRUE))
+```
+
+When used with a positional scale it acts in much the same way, placing data at the center of the bin they belong to:
+
+```{r}
+p + 
+  scale_x_binned()
+```
+
+One of the benefits of this is that it is now a breeze to create histograms with tick-marks between the bars, as a histogram is effectively a binned scale with a bar geom:
+
+```{r}
+ggplot(mpg) + 
+  geom_bar(aes(displ)) + 
+  scale_x_binned()
+```
+
+### Bi-directional geoms and stats
+While we are messing with the foundation of ggplot2, we might as well challenge another paradigm, namely that certain geoms have a direction, and to change the direction you'd have to use `coord_flip()`. The prime example of this is a horizontal bar chart:
+
+```{r}
+ggplot(mpg) + 
+  geom_bar(aes(x = manufacturer)) + 
+  coord_flip()
+```
+
+This approach has served ggplot2 well, and will continue to work in the future, but with this release we update all the directional stats and geoms to work in both directions. The direction of the stat/geom is deduced from the aesthetic mapping, so it should simply behave as expected. The example above can thus be rewritten to:
+
+```{r}
+ggplot(mpg) + 
+  geom_bar(aes(y = manufacturer))
+```
+
+The direction determination is not only looking at which aesthetics are mapped, but also what they are mapped to, so most layers can be determined without ambiguity. In the presence of ambiguity it is conservative and will default to the standard orientation. If you want to overwrite the direction, either because it fails to detect it, or because the geom/stat is ambiguous by nature, It can be set directly with the `orientation` argument:
+
+```{r, message=FALSE}
+ggplot(mpg, aes(displ, hwy)) + 
+  geom_point() + 
+  geom_smooth(orientation = "y")
+```
+
+### More control over aesthetic evaluation
+When mapping data to aesthetics we are usually differentiating between two different mappings. One for the stat and one for the geom. We don't think about this that often, because the stat simply accept the mappings and carry them over to the geom (sometimes with modifications). Sometimes, though, we want to control the mapping that happens between the output of the stat and the geom. The approach to this has been to use `stat()` (and `..var..` notation before that), to indicate that the aesthetic should be evaluated after the stat has been computed.
```

Comment:
```suggestion
When mapping data to aesthetics we are usually differentiating between two mappings: one for the stat, and one for the geom. We don't think about this that often, because the stat simply accepts the mappings and carries them over to the geom (sometimes with modifications). Sometimes, though, we want to control the mapping that happens between the output of the stat and the geom. The approach to this has been to use `stat()` (and `..var..` notation before that), to indicate that the aesthetic should be evaluated after the stat has been computed.
```

Reply:
Refactored that first sentence a bit, and fixed noun-verb agreement